src/main/java/com/redhat/exhort/utils/PythonControllerBase.java

```diff
@@ -402,42 +434,39 @@ private void bringAllDependencies(
                   + " virtual environment ( will slow down the analysis)",
               depName));
     }
-    String depVersion = getDependencyVersion(record);
-    List<String> directDeps = getDepsList(record);
-    getDependencyNameShow(record);
-    Map<String, Object> entry = new HashMap<String, Object>();
-    dependencies.add(entry);
-    entry.put("name", getDependencyNameShow(record));
-    entry.put("version", depVersion);
-    List<Map<String, Object>> targetDeps = new ArrayList<>();
-    directDeps.stream()
-        .forEach(
-            dep -> {
-              if (!path.contains(dep.toLowerCase())) {
-                List<String> depList = new ArrayList();
-                depList.add(dep.toLowerCase());
-
-                if (includeTransitive) {
-                  bringAllDependencies(
-                      targetDeps,
-                      dep,
-                      cachedTree,
-                      includeTransitive,
-                      Stream.concat(path.stream(), depList.stream()).collect(Collectors.toList()));
-                }
-              }
-              Collections.sort(
-                  targetDeps,
-                  (o1, o2) -> {
-                    String string1 = (String) (o1.get("name"));
-                    String string2 = (String) (o2.get("name"));
-                    return Arrays.compare(string1.toCharArray(), string2.toCharArray());
-                  });
-              entry.put("dependencies", targetDeps);
-            });
+
+    Map<String, Object> dataMap = new HashMap<>();
+    dataMap.put("name", pythonDependency.getName());
+    dataMap.put("version", pythonDependency.getVersion());
+    dependencyList.add(dataMap);
+
+    List<Map<String, Object>> transitiveDepList = new ArrayList<>();
+    List<String> directDeps = pythonDependency.getDependencies();
+    for (String directDep : directDeps) {
+      if (!path.contains(directDep.toLowerCase())) {
+        List<String> depList = new ArrayList<>();
+        depList.add(directDep.toLowerCase());
+
+        if (includeTransitive) {
+          bringAllDependencies(
+              transitiveDepList,
+              directDep,
+              cachedTree,
+              true,
```

Comment:
@JudeNiroshan This should be parameterized and never hard-coded `true`, as this value depends on whether the python-pip provider asking for component analysis with direct ones only ( and it asks includeTransitive=false) and for stack analysis it's asking for ncludeTransitive=true, anyway that parameter set by provider should be sent all the way deep down in the recurrsion in all the recursive calls, see below suggested change.

Reply:
Yes, you're right. But as we have the `if()` wrapped in this recursive function call, we can hard-code this to `true`. In the case of component analysis, we are not calling this recursive function as if condition will not become true.